analyzer_test.go

```diff
@@ -63,6 +65,14 @@ func testAnalyzer(t *testing.T, when spec.G, it spec.S) {
 			},
 			LayersDir: layerDir,
 			Logger:    &log.Logger{Handler: &discard.Handler{}},
+			LayerAnalyzer: lifecycle.NewLayerAnalyzer(
+				&log.Logger{Handler: &discard.Handler{}},
```

Comment:
When we construct the `LayerAnalyzer`, do we give it the same logger that `Analyzer` has? It might be good to mimic that in the test setup.

Reply:
Though in this case since it's being discarded I don't know if we care. This bit me on the detector refactor I think.